sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldNotBeIgnored.NUnit.cs

```diff
@@ -14,51 +17,95 @@ public void Foo1()
         }
 
         [Test]
-        [Microsoft.VisualStudio.TestTools.UnitTesting.Ignore] // This test is ignored because 'blah blah'
+        [NUnit.Framework.Ignore] // This test is ignored because 'blah blah'
         public void Foo2()
         {
         }
 
         [Test]
-        [Microsoft.VisualStudio.TestTools.UnitTesting.Ignore]
-        [WorkItem(1234)]
+        [Ignore]
+        [MSTest.WorkItem(1234)]
         public void Foo3()
         {
         }
 
         [TestCase("")]
-        [Microsoft.VisualStudio.TestTools.UnitTesting.Ignore]
-//       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Noncompliant
+        [Ignore()]
+//       ^^^^^^^^ Noncompliant
         public void Foo4(string s)
         {
         }
 
         [TestCase("")]
-        [Microsoft.VisualStudio.TestTools.UnitTesting.Ignore] // This test is ignored because 'blah blah'
+        [NUnit.Framework.Ignore] // This test is ignored because 'blah blah'
         public void Foo5(string s)
         {
         }
 
         [TestCase("")]
-        [Microsoft.VisualStudio.TestTools.UnitTesting.Ignore]
-        [WorkItem(1234)]
+        [Ignore]
+        [MSTest.WorkItem(1234)]
         public void Foo6(string s)
         {
         }
 
         [TestCaseSource("DivideCases")]
-        [Microsoft.VisualStudio.TestTools.UnitTesting.Ignore]
-//       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Noncompliant
+        [Ignore]
+//       ^^^^^^ Noncompliant
         public void DivideTest(int n, int d, int q)
         {
             Assert.AreEqual(q, n / d);
         }
 
         static object[] DivideCases =
         {
-            new object[] { 12, 3, 4 },
-            new object[] { 12, 2, 6 },
-            new object[] { 12, 4, 3 }
+            new object[] { 12, 3, 4 }
         };
+
+        [Datapoint]
+        public int Data = 42;
+
+        [Theory]
+        [Ignore]
+//       ^^^^^^ Noncompliant
+        public void Theory1(int arg)
+        {
+        }
+
+        [Theory]
+        [Ignore] // some reason
+        public void Theory2(int arg) // Compliant
+        {
+        }
+
+        [Theory]
+        [Ignore("a reason")]
+        public void Theory3(int arg)
+        {
+        }
+    }
+
+    [TestFixture, Ignore]
+    public class IgnoredClass1
```

Comment:
Added tests for class-level NUnit _Ignore_ (same as corresponding MSTest class-level tests).